app/services/offer_authorisation.rb

```diff
@@ -0,0 +1,74 @@
+class OfferAuthorisation
+  def initialize(actor:)
+    @actor = actor
+  end
+
+  def can_make_offer?(application_choice:, course_option_id:)
```

Comment:
What if we passed all the params in the constructor and then asked `granted?` or something? To remove the redundancy from the name

```ruby
authorisation = MakeOfferAuthorisation.new(actor: x, application_choice: y, course_option_id: z)
if authorisation.granted?
  do_stuff
end
```

Reply:
I didn't want to refactor too much for this, it's likely to change anyway. Okay if we look at this later? There's a chance that this goes back into the class anyway, or that we extract a bunch of other `XAuthorisation` classes.